src/grafana_agent.py

```diff
@@ -210,58 +195,58 @@ def on_remote_write_changed(self, event: RelationChangedEvent) -> None:
 
     def _update_status(self) -> None:
         """Update the status to reflect the status quo."""
-        if len(self.model.relations["metrics-endpoint"]):
-            if not len(self.model.relations[REMOTE_WRITE_RELATION_NAME]):
-                self.unit.status = WaitingStatus("no related Prometheus remote-write")
+        if len(self.charm.model.relations["metrics-endpoint"]):
+            if not len(self.charm.model.relations[REMOTE_WRITE_RELATION_NAME]):
+                self.charm.unit.status = WaitingStatus("no related Prometheus remote-write")
                 return
 
-        if not self.unit.get_container("agent").can_connect():
-            self.unit.status = WaitingStatus("waiting for the agent container to start")
+        if not self._is_ready():
+            self.charm.unit.status = WaitingStatus("waiting for the agent to start")
             return
 
-        self.unit.status = ActiveStatus()
+        self.charm.unit.status = ActiveStatus()
 
     def _update_config(self, _) -> None:
-        if not self._container.can_connect():
-            # Pebble is not ready yet so no need to update config
-            self.unit.status = WaitingStatus("waiting for agent container to start")
+        if not self._is_ready():
+            # Grafana-agent is not yet running so no need to update config
+            self.charm.unit.status = WaitingStatus("waiting for agent to start")
```

Comment:
so now we have an object other than the charm manipulating the charm's status? Seems like a risky pattern to me.

Reply:
see leading comment for the context